src/Orleans/Core/GrainIdentity.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using Orleans.Runtime;
+
+namespace Orleans.Core
+{
+    internal class GrainIdentity : IGrainIdentity
```

Comment:
Agree. Good plan. So we make GrainId  implement  IGrainIdentity but leave  GetGrainId()  and  GetGrainIdentity()  of  GrainExtensions and remember to refactor  GrainId  in the future to only implement  IGrainIdentity  and not all the rest. I need to look closely what is that all the rest.